opts/network.go

```diff
@@ -0,0 +1,77 @@
+package opts
+
+import (
+	"encoding/csv"
+	"fmt"
+	"github.com/docker/docker/api/types/swarm"
+	"strings"
+)
+
+const (
+	networkOptName  = "name"
+	networkOptAlias = "alias"
+	networkOptIP    = "ip"
+	networkOptIPv6  = "ipv6"
+)
+
+// NetworkOpt represents a network config in swarm mode.
+type NetworkOpt struct {
+	networkAttachments []swarm.NetworkAttachmentConfig
+}
+
+// Set networkopts value
+func (n *NetworkOpt) Set(value string) error {
+	csvReader := csv.NewReader(strings.NewReader(value))
+	fields, err := csvReader.Read()
+	if err != nil {
+		return err
+	}
+	var netAttach swarm.NetworkAttachmentConfig
+	for _, field := range fields {
+		parts := strings.SplitN(field, "=", 2)
+		//Support legacy non csv format
+		if len(parts) == 1 {
+			netAttach.Target = parts[0]
+			break
+		}
+
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid field %s", field)
```

Comment:
@thaJeztah I was. following the error message https://github.com/moby/moby/blob/master/opts/port.go#L43